Gcpe.Hub.API/Controllers/ActivitiesController.cs

```diff
@@ -0,0 +1,161 @@
+    public class ActivitiesController : ControllerBase
+    {
+        private readonly HubDbContext dbContext;
+        private readonly ILogger<ActivitiesController> logger;
+        private readonly IMapper mapper;
+
+        public ActivitiesController(HubDbContext dbContext,
+            ILogger<ActivitiesController> logger,
+            IMapper mapper)
+        {
+            this.dbContext = dbContext;
+            this.logger = logger;
+            this.mapper = mapper;
+        }
+
+        [HttpGet("Forecast/{numDays}")]
+        [Produces(typeof(IEnumerable<ActivityViewModel>))]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(400)]
+        public IActionResult GetNewsForecast(int numDays)
+        {
+            try
+            {
+                var today = new DateTime(2018, 9, 1);// DateTime.Today;
+                IList<ActivityViewModel> forecast = dbContext.Activity.Include(a => a.ContactMinistry).Include(a => a.City)
+                    .Include(a => a.ActivityCategories).ThenInclude(ac => ac.Category)
+                    .Include(a => a.ActivitySharedWith).ThenInclude(sw => sw.Ministry)
+                    .Where(a => a.StartDateTime >= today && a.StartDateTime <= today.AddDays(numDays) && !a.IsConfidential && a.IsConfirmed && a.IsActive &&
+//                               a.ActivityKeywords.Any(ak => ak.Keyword.Name.StartsWith("HQ-")) &&
+                               a.ActivityCategories.Any(ac => ac.Category.Name.StartsWith("Approved") || ac.Category.Name == "Release Only (No Event)" || ac.Category.Name.EndsWith("with Release")))
+                    .Select(a => mapper.Map<Activity, ActivityViewModel>(a)).ToList();
+
+                return Ok(forecast);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError($"Failed to get activities: {ex}");
+                return BadRequest("Failed to get activities");
+            }
+        }
+
+        [HttpGet("{id}")]
+        [Produces(typeof(ActivityViewModel))]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(400)]
+        [ProducesResponseType(404)]
+        public IActionResult GetById(string id)
+        {
+            try
+            {
+                var activity = dbContext.Activity.Find(id);
+
+                if (activity != null)
+                {
+                    return Ok(mapper.Map<Activity, ActivityViewModel>(activity));
+                }
+                else return NotFound();
+            }
+            catch (Exception ex)
+            {
+                logger.LogError($"Failed to get activity: {ex}");
+                return BadRequest("Failed to get activity");
+            }
+        }
+
+        [HttpPost]
+        [ProducesResponseType(201)]
+        [ProducesResponseType(400)]
+        public IActionResult Post([FromBody]ActivityViewModel activity)
+        {
+            try
+            {
+                if (ModelState.IsValid)
```

Comment:
The [ApiController] on the controller allows us to avoid explicitly checking model state: https://www.strathweb.com/2018/02/exploring-the-apicontrollerattribute-and-its-features-for-asp-net-core-mvc-2-1/